### ISSUE TITLE
table: filter icon does not show whether filter is active on column

### ISSUE THREAD

**Issue by dralucard666** [primefaces/primeng#18030]
### Describe the bug

Users cannot easily see which columns are filtered. We need the filtericon template to include context about the filter's state to improve usability and accessibility.

![Image](https://github.com/user-attachments/assets/5cc2ff4a-1e85-4817-bb96-6c2e676b6da8)

(should be possible to make icon look like column 'company')

At the moment, the ColumnFilter component does not provide context to the filterIconTemplate. Hence it is not possible to switch icons for both states.
Primes current approach of adding a second template for removing the filter ('clearFilterIconTemplate') is not compatible with our style guideline.

### Pull Request Link

https://github.com/primefaces/primeng/pull/18029

### Reproducer

https://stackblitz.com/edit/github-c9pcpcr4?file=src%2Fapp%2Fapp.component.html,src%2Fapp%2Fapp.component.ts

### Environment

--

### Angular version

19.1

### PrimeNG version

v19

### Node version

20.18

### Steps to reproduce the behavior

1. Open example on stackblitz.
2. Filter icon for column "name" stays the same whether a filter is set or not.

### Expected behavior

We would want the icon to look like column "Company" if a filter is active.
At the moment this is not (or hardly) possible with the current implementation.
Hence we propose adding context to the filterIconTemplate so that we can switch the icons within our own template.
If this should be the default case, without adding the custom template, we can also change that in the pr.

**Comment by mehmetcetin01140** [comment]
fixed in ab68391b8b9dd2f3b927d392c6e17715468bd363
